aesara/tensor/math_opt.py

```diff
@@ -219,6 +221,19 @@ def local_func_inv(fgraph, node):
 
     for inv_pair in inv_pairs:
         if is_inverse_pair(node_op, prev_op, inv_pair):
+
+            # Special case for exp(log(x))
+            if isinstance(prev_op, aes.Log):
+                x = x.owner.inputs[0]
+                new_out = switch(ge(x, 0), x, np.nan)
+
+                old_out = node.outputs[0]
+                if new_out.dtype != old_out.dtype:
+                    new_out = cast(new_out, dtype=old_out.dtype)
+                if new_out.type != old_out.type:
+                    return
+                return [new_out]
```

Comment:
This is supposed to be a generalized implementation, so let's not hard-code special cases.

Reply:
I see the point, I was trying to avoid checking ever pair of nodes twice in different functions.